### ISSUE TITLE
Can't set brightness of lights in a zone without specifying the light name

### ISSUE THREAD

**Issue by Sleuth56** [home-assistant/intents#2763]
Steps to reproduce:
1. Setup voice pipline
2. Add a smart light to a zone
3. Add a voice device to HA and set it to the same zone as the light
4. Say to the voice device `Set the lights to 50%`

![Screenshot_20241219-180318](https://github.com/user-attachments/assets/f4fa3b9d-81a6-44e5-9bef-4806de4fde3b)

I did notice in my testing that `Set the brightness to 50%` works.
![Screenshot_20241219-180927](https://github.com/user-attachments/assets/ad79b4e6-affb-4455-8b0d-1601a50c7978)

**Comment by mitrokun** [comment]
There was a similar question in this [issue](https://github.com/home-assistant/intents/issues/2237) 
This feature is easy to implement, but is it worth it?

**Comment by ViViDboarder** [comment]
Should be covered by #2783

**Comment by tetele** [comment]
> This feature is easy to implement, but is it worth it?

Yes, it is. I can imagine sitting in a room with dimmable lights, asking for the lights to be set to X% brightness. Of course i mean just that room.

**Comment by mitrokun** [comment]
> I can imagine sitting in a room with dimmable lights


It should be clarified that this function was added for the English language almost a year ago  (in your PR😀) . The problem discussed in this issue is related to the lack of a short version of the phrase.

But as for my opinion, I do not think that the current implementation of this intent works well for everyone. 
If you have a large variety of lighting fixtures in the room, then running this command will lead to chaos.
Also short version of phrases may be misidentified if local tiny/small stt models are used.

A solution could be to add to the HA the functionality of dividing the light in the room into main and auxiliary.
Then the phrase "Set lights to 50%" would only affect the main light, and "Set all lights to 50%" would affect all devices in the room.  Auxiliary lights are controlled by name.
In this case, false alarms would not be so critical, and the convenience of controlling lighting devices would increase.
Now this problem is solved by creating additional zones or using shortcuts and own intent_script

The same questions apply to the intentions of turning on the light.

**Comment by sappafrancesco** [comment]
Looked into why "set the lights to 50%" (no name/area) doesn't work while "set the brightness to 50%" does.

It's not a sentence-template bug, it's a schema gap. `HassLightSet` in `intents.yaml` only declares combinations that require `name`, `area`, or `floor`:

```
name_brightness, name_color, name_temperature,
area_brightness, area_color, area_temperature,
floor_brightness, floor_color, floor_temperature
```

There's no bare/context-area combination (`slots: [brightness]` with `context_area: true`), unlike e.g. `HassMediaSearchAndPlay`'s `default` combo, which uses exactly that pattern to mean "the media player in the voice satellite's area" without naming anything:

```yaml
default:
  description: "Search and play on the media player in the voice satellite's area"
  importance: "optional"
  slots:
    - "search_query"
  context_area: true
```

Adding an equivalent combo to `HassLightSet` (say `area_context_brightness`, slots `[brightness]`, `context_area: true`) would be enough to make "set the lights to 50%" resolve to whatever area the voice satellite is in, mirroring the media intent. But `intents.yaml`'s slot_combinations are maintainer-only (enforced by the `guard-core-files` CI check), so this isn't something a language contributor can add directly. It needs a maintainer to add the combo to the schema; once that exists, wiring it up per-language (sentences, tests, responses) is a normal contribution.

Flagging this as the actual blocker in case a maintainer wants to pick it up, since the fix is small and there's already a working precedent to copy from.
